.github/workflows/scripts/pr-tests/.pinot_tests_unit.sh

```diff
@@ -29,24 +29,24 @@ netstat -i
 #   - TEST_SET#1 runs install and test together so the module list must ensure no additional modules were tested
 #     due to the -am flag (include dependency modules)
 if [ "$RUN_TEST_SET" == "1" ]; then
-  mvn test -T 16 \
+  mvn test jacoco:report-aggregate -T 16 \
       -pl 'pinot-spi' \
       -pl 'pinot-segment-spi' \
       -pl 'pinot-common' \
       -pl ':pinot-yammer' \
       -pl 'pinot-core' \
       -pl 'pinot-query-planner' \
       -pl 'pinot-query-runtime' \
-      -P github-actions,no-integration-tests || exit 1
+      -P github-actions,codecoverage,no-integration-tests || exit 1
```

Comment:
I don't think this is working as expected. Given we are providing a inclusive list of projects and `-pl reports` is not included, we are not actually executing the aggregate defined in reports/pom.xml. What we are doing is to directly run `jacoco:report-aggregate` on each project, which I don't think it is super useful.

We should also apply this change in the integration tests file (`.pinot_tests_integration.sh`)

Am I right?

Reply:
almost right :) The root issue is the split of modules to build/test, while the reports needs to have all modules. So the reports-module fails if dependencies are missing. By putting `jacoco:report-aggregate` on the commandline, it is execute as last (which is the opposite of current behavior: run aggregate as first). With inherited=false we can separate the root module from its child modules, but I guess I need to configure the skip correctly for both.
If this works, I suggest to remove the reports-module as it will only lead to confusion.